e2e/tests/workspace/workspace-share.spec.ts

```diff
@@ -49,5 +55,69 @@ describe('Share workspace', () => {
       accessLevel = await aboutPage.findUserInCollaboratorList(config.collaboratorUsername);
       expect(accessLevel).toBeNull();
     });
+
+
+    /**
+     * Test:
+     * - Create a new workspace.
+     * - Share with another user with role READER.
+     * - Log in as another user.
+     * - Workspace share action should be disabled.
+     */
+    test('Workspace READER cannot share edit or delete workspace', async () => {
+
+      const workspaceCard = await findWorkspace(page, true);
+      const workspaceName = await workspaceCard.getWorkspaceName();
+
+      // Open the Share modal
+      const menu = workspaceCard.getEllipsis();
+      await menu.clickAction(EllipsisMenuAction.Share, {waitForNav: false});
+
+      const shareModal = new ShareModal(page);
+      await shareModal.waitUntilVisible();
+
+      await shareModal.shareWithUser(config.collaboratorUsername, WorkspaceAccessLevel.Reader);
+      await waitWhileLoading(page);
+      await Navigation.navMenu(page, NavLink.SIGN_OUT);
+
+      // To verify READER role assigned correctly, user with READER role will sign in in new Incognito page.
+      const newPage = await signInAs(config.collaboratorUsername, config.userPassword);
+
+      const homePage = new HomePage(newPage);
+      await homePage.getSeeAllWorkspacesLink().then((link) => link.click());
+
+      const workspacesPage = new WorkspacesPage(newPage);
+      await workspacesPage.waitForLoad();
+
+      // Verify Workspace Access Level is READER.
+      const workspaceCard2 = await WorkspaceCard.findCard(newPage, workspaceName);
+      const accessLevel = await workspaceCard2.getWorkspaceAccessLevel();
+      expect(accessLevel).toBe(WorkspaceAccessLevel.Reader);
+
+      // Share, Edit and Delete actions are not available for click.
+      const menu2 = workspaceCard2.getEllipsis();
+      await menu2.clickEllipsis();
+      await menu2.isDisabled(EllipsisMenuAction.Share);
+      await menu2.isDisabled(EllipsisMenuAction.Edit);
+      await menu2.isDisabled(EllipsisMenuAction.Delete);
+
+      // Duplicate action is available for click.
+      await menu2.isDisabled(EllipsisMenuAction.Duplicate);
+
+      // Make sure the Search input-field in Share modal is disabled.
+      await workspaceCard2.clickWorkspaceName();
+      await (new DataPage(newPage)).openTab(TabLabelAlias.About);
+      const aboutPage = new WorkspaceAboutPage(newPage);
+      await aboutPage.waitForLoad();
+      const modal2 = await aboutPage.openShareModal();
```

Comment:
[no action] TIL that you can open the share modal even if you're a read-only user. Is this somehow expected / required for the product? The only thing I can think of is that this is the path for a read-only user to remove themself from the workspace... but it does feel like an awkward flow if that's the case. More common / expected (e.g. for things like Google Groups) would be a button that says "Remove me from this workspace".

Reply:
I thought it was odd to find enabled Share button in About page. Maybe the purpose of Share modal is to show information about this workspace who else are OWNER/READER/WRITER.